File: src/main/scala/com/vc/L700.scala
package com.vc

class TreeNode700(var _value: Int) {
  var value: Int = _value
  var left: TreeNode700 = null
  var right: TreeNode700 = null
}

object L700 {
  def searchBST(root: TreeNode700, value: Int): TreeNode700 = {
    var current = root
    while(current != null) {
      if(current.value < value) {
        current = current.right
      }
      else if(current.value > value) {
        current = current.left
      }
      else return current
    }
    current
  }
}

File: src/main/java/com/vc/easy/L700.java
package com.vc.easy;

import com.vc.hard.TreeNode;

class L700 {
    public TreeNode searchBST(TreeNode root, int val) {
        TreeNode current = root;
        while(current != null) {
            if(current.val > val) current = current.left;
            else if(current.val < val) current = current.right;
            else return current;
        }
        return null;
    }
}
